apps/src/JSInterpreter.js

```diff
@@ -372,13 +372,17 @@ JSInterpreter.prototype.executeInterpreter = function (firstStep, runUntilCallba
     if ((reachedBreak && !unwindingAfterStep) ||
         (doneUserLine && !unwindingAfterStep && !atMaxSpeed) ||
         this.yieldExecution ||
+        this.interpreter.paused_ ||
```

Comment:
@joshlory - no, because `yieldExecution` only yields from the current `executeInterpreter` loop. It is currently used in 2 places. Once, when the `nativeGetCallback()` function detects an empty `eventQueue` (I remember we were looking for where this occurred the other day!). And secondly, in Play Lab when we decouple interpreter execution ticks from animation ticks (a mode only used in star wars levels 1-6)